### PR TITLE
Zt 101 rma sale fix#4

### PR DESCRIPTION
**Pull request by zatinrma**
https://trello.com/c/MmhZ8XIc/235-action-button-create-po-in-form-view-vsb
https://trello.com/c/f3Y1QD8Y/237-fix-field-delivery-to-in-wizard-form-create-po-from-vsb

### CODE REVIEW

File: rma_vsb/wizard/rma_vsb_order_wizard.py [hunk 1]
```diff
@@ -18,15 +20,35 @@ class RmaVsbOrderWizard(models.TransientModel):
...
+    company_id = fields.Many2one('res.company', required=True, readonly=True, default=lambda self: self.env.company)
 
     @api.model
     def default_get(self, fields):
         res = super(RmaVsbOrderWizard, self).default_get(fields)
         res['date_order'] = datetime.now()
-        if self._context['vsb_order_line']:
-            res['vsb_order_line_ids'] = self._context['vsb_order_line']
+        # For user create PO from 1 rec of VSB form view
+        if self._context.get('create_po_from_one_vsb_rec'):
```
Comment (panharithrma) [inline rma_vsb/wizard/rma_vsb_order_wizard.py:33]: no need to use context, I think

File: rma_vsb/wizard/rma_vsb_order_wizard.py [hunk 2]
```diff
@@ -18,15 +20,35 @@ class RmaVsbOrderWizard(models.TransientModel):
...
+            selected_vsb_ids = self.env['rma.vsb.purchase.tracking'].browse(self._context.get('active_ids'))
+            # run validation
+            if selected_vsb_ids:
+                vals_prod = []
+                for vsb_line in selected_vsb_ids:
+                    if not vsb_line.lot_serial_number or not vsb_line.engine:
```
Comment (panharithrma) [inline rma_vsb/wizard/rma_vsb_order_wizard.py:39]: use function all() to increase performance and cleaner code

**Review by panharithrma — CHANGES_REQUESTED** [review]

File: rma_vsb/wizard/rma_vsb_order_wizard.py (hunk 1 above)
Comment (zatinrma) [reply rma_vsb/wizard/rma_vsb_order_wizard.py:33]: 2 Different Action

File: rma_vsb/wizard/rma_vsb_order_wizard.py [hunk 3]
```diff
@@ -36,14 +56,20 @@ def action_create_po_from_vsb_wizard(self):
...
 
-        for line in self._context['vsb_order_line']:
+        # For user create PO from 1 rec of VSB form view, No need validation since we already validate in default get
+        if self._context.get('create_po_from_one_vsb_rec'):
+            selected_vsb_ids = self.env['rma.vsb.purchase.tracking'].browse(self._context.get('active_ids'))
+            vsb_order_line = selected_vsb_ids.extract_order_line(selected_vsb_ids)
+            # Update the context to the multiple vsb flow
+            self = self.with_context(vsb_order_line=vsb_order_line)
+
+        for line in self._context.get('vsb_order_line'):
```
Comment (panharithrma) [inline rma_vsb/wizard/rma_vsb_order_wizard.py:72]: change to self._context.get('vsb_order_line', [])

File: rma_vsb/wizard/rma_vsb_order_wizard.py [hunk 4]
```diff
@@ -53,7 +79,7 @@ def action_create_po_from_vsb_wizard(self):
             'default_date_order': self.date_order,
             'default_date_planned': self.date_planned,
             'default_picking_type_id': self.picking_type_id.id if self.picking_type_id else False,
-            'default_order_line': self._context['vsb_order_line'],
+            'default_order_line': self._context.get('vsb_order_line'),
```
Comment (panharithrma) [inline rma_vsb/wizard/rma_vsb_order_wizard.py:82]: change to self._context.get('vsb_order_line', [])

**Review by panharithrma — CHANGES_REQUESTED** [review]

**Pull request merged**